web/src/modals_util.ts

```diff
@@ -19,6 +19,16 @@ export const custom_time_unit_values = {
     },
 };
 
+export function valid_custom_time_input(input: number): boolean {
+    if (input < 0) {
+        return false;
+    }
+    if (Number.isInteger(input)) {
```

Comment:
This does not handle cases where the text is something like "11a", which is considered as valid. We do similar checks for the custom inputs for message edit time limits in `settings_components.check_valid_number_input`.